src/clj/game/core-io.clj

```diff
@@ -81,10 +81,15 @@
 
 
 ; In-game chat commands
+(defn set-adv-counter [state side target value]
+  (set-prop state side target :advance-counter value)
+  (trigger-event state side :advancement-placed target)
+  (system-msg state side (str "sets advancement counters to " value " on "
```

Comment:
Triggering an event notifies all listening cards one at a time. Each card's ability is resolved before moving to the next. "Resolving" an ability either causes an immediate change to the logical game state (what the player considers to be "the state of the game": credits, cards, points, etc.), or inserts a prompt into a queue with instructions on what to do when the prompt is handled by the player. 

The "update loop" equivalent is: player performs some action on client -> message sent to Node server -> message forwarded to game server -> message parsed, game actions invoked, new game state calculated -> state returned to Node server -> state returned to clients, which render the new state.